rootfs/network/default.script

```diff
@@ -57,7 +57,7 @@ case "$1" in
 
         if [ -n "$hostname" ]; then
             hostname "$hostname"
-            echo Hostname set to $(hostname)
+            echo Hostname set to $(hostname), supplied by DHCP server
         elif [ -e /etc/hostname ]; then
             hostname "$(cat /etc/hostname)"
         elif [ "$(hostname -s)" == "panda" ]; then
```

Comment:
Not really part of this review, but what's this speclal case for `` `hostname` = panda`` about?  It looks as if it might interact with what you're trying to do in this commit.

Reply:
`panda` is the default hostname. The idea is that if `HOSTNAME` is not specified in `config.txt` and `-O hostname` doesn't give us the hostname from the DHCP server (it doesn't seem to work at DLS) then it will try and do a reverse lookup from DNS.

It would be better to do `! HOSTNAME="$(parse_config)"` here, but that function is in a different file...

Reply:
I also don't know what (if anything) sets `/etc/hostname`, I can't see it mentioned anywhere else

Reply:
Oh, that's interesting.  Is `/etc/hostname` even present in the rootfs build?  If not, it can't be created afterwards... (I don't have a build to hand to check).

If not, it looks like all references to this file are relics of alternative versions of the scripts.  Both `default.script` and `network` (in `rootfs/network`) could probably be edited to drop all references to `/etc/hostname`.

Reply:
```
root@panda-tst:~# ls /etc/hostname
ls: /etc/hostname: No such file or directory
root@panda-tst:~# touch /etc/hostname
touch: /etc/hostname: Read-only file system
```

>  Both default.script and network (in rootfs/network) could probably be edited to drop all references to /etc/hostname.

Ok, I'll do that